src/main/java/org/sagebionetworks/bridge/AuthUtils.java

```diff
@@ -235,4 +250,45 @@ public static boolean isInRole(Set<Roles> callerRoles, Set<Roles> requiredRoles)
         return callerRoles != null && requiredRoles != null && 
                 requiredRoles.stream().anyMatch(role -> isInRole(callerRoles, role));
     }
+    
+    
+    /**
+     * To access an individual account, one of these conditions must hold true:
+     * 
+     * 1) it's the caller's account;
+     * 2) the account is enrolled in a study and the caller is a researcher;
+     * 3) the account is enrolled in a study and the caller is a study coordinator 
+     *    with access to that study;
+     * 4) the account is a test account and the caller is a developer;
+     * 5) the account is a test account enrolled in a study and the caller is a 
+     *    study coordinator with access to that study.
+     *    
+     * THIS METHOD HAS AN IMPORTANT SIDE EFFECT: it may remove enrollments from the 
+     * supplied account that are not visible to the caller, so it can be returned to 
+     * the caller without leaking enrollment information. Do not call this method 
+     * before updating an account record.
+     */
+    public static final boolean canAccessAccount(Account account) {
+        if (account == null) {
+            return false;
+        }
+        Set<String> userDataGroups = account.getDataGroups();
+        if (IS_ONLY_DEVELOPER.check(USER_ID, account.getId()) && !userDataGroups.contains(TEST_USER_GROUP)) {
+            return false;
+        }
+        // If this is a call for one’s own record, or the caller is an admin or worker, or the account 
+        // is in the same organization as the caller who is an org admin, return the account.
+        if (CAN_READ_PARTICIPANTS.check(USER_ID, account.getId(), ORG_ID, account.getOrgMembership())) {
+            return true;
+        }
+        // If after removing all enrollments that are not visible to the caller, there are no remaining 
+        // enrollments, the caller cannot access the account.
+        RequestContext context = RequestContext.get();
+        Set<String> callerStudies = context.getOrgSponsoredStudies();
+        Set<Enrollment> removals = account.getEnrollments().stream()
+                .filter(en -> !callerStudies.contains(en.getStudyId()))
+                .collect(toSet());
+        account.getEnrollments().removeAll(removals);
+        return !account.getEnrollments().isEmpty();
```

Comment:
An effect here would be that developers not associated to an organization would not be able to see test accounts, because they have no relationship to the study.

Reply:
No, not true. A developer passes the AuthUtils check above as a developer. It never gets to that check if the account is not a test account. The end result is that developers can access all test accounts in an app.